addons/product/models/product_template.py

```diff
@@ -28,6 +28,10 @@ def _get_default_uom_id(self):
         # Deletion forbidden (at least through unlink)
         return self.env.ref('uom.product_uom_unit')
 
+    @tools.ormcache()
+    def _get_default_uom_po_id(self):
+        return self.env['uom.uom'].browse(self.default_get(['uom_id']).get('uom_id')) or self.env.ref('uom.product_uom_unit')
```

Comment:
```suggestion
    def _get_default_uom_po_id(self):
        return self.default_get(['uom_id']).get('uom_id') or self._get_default_uom_id()
```

The method should not be cached as it depends on the environment (user & context).
The returned value doesn't have to be converted to a record, an integer is also supported.
Since `ref` costs one SQL query (to check the record existence), you might as well reuse the cached method `_get_default_uom_id` to get the default unit uom :).